tests/testthat/test-derive_param_tte.R

```diff
@@ -932,17 +932,203 @@ test_that("derive_param_tte Test 13: error if dataset_name not in source_datsets
+test_that("derive_param_tte Test 15: derive_param_tte resolves ties using order argument when
+input is sorted descending", {
+  adsl <- tibble::tribble(
+    ~USUBJID, ~TRTSDT,           ~EOSDT,
+    "01",     ymd("2020-12-06"), ymd("2021-03-06"),
+    "02",     ymd("2021-01-16"), ymd("2021-02-03")
+  ) %>%
+    mutate(STUDYID = "AB42")
+
+  # Sort the input AE dataset in descending order by AESEQ
+  # to confirm that the order argument re-sorts it correctly.
+  ae <- tibble::tribble(
+    ~USUBJID, ~AESTDTC, ~AESEQ, ~AESER, ~AEDECOD,
+    "01", "2021-01-03", 2, "Y", "Cough",
+    "01", "2021-01-03", 1, "Y", "Flu",
+    "01", "2021-01-20", 3, "N", "Headache"
+  ) %>%
+    mutate(
+      STUDYID = "AB42",
+      AESTDT = ymd(AESTDTC)
+    ) %>%
+    arrange(desc(AESEQ)) # Intentionally sort descending to test the order argument
+
+  result <- derive_param_tte(
+    dataset_adsl = adsl,
+    start_date = TRTSDT,
+    event_conditions = list(event_source(
+      dataset_name = "ae",
+      date = AESTDT,
+      set_values_to = exprs(
+        EVENTDESC = "Serious AE",
+        SRCSEQ = AESEQ
+      ),
+      filter = AESER == "Y",
+      order = exprs(AESEQ) # Should re-sort so that AESEQ=1 (Flu) is chosen on tie
+    )),
+    censor_conditions = list(censor_source(
+      dataset_name = "adsl",
+      date = EOSDT,
+      censor = 1,
+      set_values_to = exprs(EVENTDESC = "End of Study")
+    )),
+    set_values_to = exprs(
+      PARAMCD = "TTSAE",
+      PARAM = "Time to First Serious AE"
+    ),
+    source_datasets = list(adsl = adsl, ae = ae)
+  )
+
+  # Check that for USUBJID = "01", the first serious AE selected is the one with AESEQ = 1 (Flu),
+  # despite the input AE data initially being arranged to show AESEQ=2 (Cough) first.
+  selected_seq <- result %>%
+    filter(USUBJID == "01", PARAMCD == "TTSAE") %>%
+    pull(SRCSEQ)
+
+  expect_equal(selected_seq, 1, info = "The order argument should ensure AE with AESEQ=1
+  is chosen on tie.")
+})
+
+## Test 16: derive_param_tte produces consistent results regardless of input sort order ----
+test_that("derive_param_tte Test 16: produces consistent results regardless of input sort order", {
+  # Define ADSL dataset
+  adsl <- tibble::tribble(
+    ~USUBJID, ~TRTSDT,           ~TRTEDT,           ~EOSDT,
+    "01",     ymd("2020-12-06"), ymd("2021-03-02"), ymd("2021-03-06"),
+    "02",     ymd("2021-01-16"), ymd("2021-01-20"), ymd("2021-02-03")
+  ) %>% mutate(STUDYID = "AB42")
+
+  # Define AE dataset with duplicates
+  ae <- tibble::tribble(
+    ~USUBJID, ~AESTDTC,     ~AESEQ, ~AEDECOD,
+    "01",     "2021-01-03",      1, "Flu",
+    "01",     "2021-03-04",      2, "Cough",
+    "01",     "2021-01-03",      3, "Flu"
+  ) %>% mutate(STUDYID = "AB42", AESTDT = ymd(AESTDTC))
+
+  # Deduplicate AE dataset to remove duplicate warnings
+  ae <- ae %>% distinct(STUDYID, USUBJID, AESTDT, .keep_all = TRUE)
```

Comment:
Why are the duplicates (with respect to the date) removed?

I think without the duplicates the test doesn't make sense because the results were affected by the sort order only when duplicates occurred. Or did I miss something?